src/zenml/integrations/kubeflow/docker_utils.py

```diff
@@ -62,14 +81,73 @@ def create_dockerfile(
     if command:
         lines.append(f"CMD {command}")
 
-    if not output_path:
-        _, output_path = tempfile.mkstemp()
+    return "\n".join(lines)
 
-    logger.debug("Writing dockerfile to path '%s'", output_path)
-    with open(output_path, "w") as output_file:
-        output_file.write("\n".join(lines))
 
-    return output_path
+def create_custom_build_context(
+    build_context_path: str,
+    dockerfile_contents: str,
+    dockerignore_path: Optional[str] = None,
+) -> Any:
+    """Creates a docker build context.
+
+    Args:
+        build_context_path: Path to a directory that will be sent to the
+            docker daemon as build context.
+        dockerfile_contents: File contents of the Dockerfile to use for the
+            build.
+        dockerignore_path: Optional path to a dockerignore file. If no value is
+            given, the .dockerignore in the root of the build context will be
+            used if it exists.
+
+    Returns:
+        Docker build context that can be passed when building a docker image.
+    """
+    exclude_patterns = []
+    default_dockerignore_path = os.path.join(
+        build_context_path, ".dockerignore"
+    )
+    if dockerignore_path:
+        exclude_patterns = _parse_dockerignore(dockerignore_path)
+    elif fileio.file_exists(default_dockerignore_path):
+        logger.info(
+            "Using dockerignore found at path '%s' to create docker "
+            "build context.",
+            default_dockerignore_path,
+        )
+        exclude_patterns = _parse_dockerignore(default_dockerignore_path)
+
+    logger.debug(
+        "Exclude patterns for creating docker build context: %s",
+        exclude_patterns,
+    )
+    no_ignores_found = not exclude_patterns
+
+    files = docker_build_utils.exclude_paths(
+        build_context_path, patterns=exclude_patterns
+    )
+    extra_files = [("Dockerfile", dockerfile_contents)]
+    context = docker_build_utils.create_archive(
+        root=build_context_path,
+        files=sorted(files),
+        gzip=False,
+        extra_files=extra_files,
+    )
+
```

Comment:
i would add an else with a log that you did not specify  a `dockerignore` and a `.dockerignore` does not exist at root, so we are building all the context with all the files